### PR TITLE
[1.1.17]ECM Support file download

### PR DESCRIPTION
**Pull request by v-kkhuang**
### What is the purpose of the change

ECM Support file download

### Related issues/PRs

Related issues:  #4949

### Brief change log

- Define the core abstraction and interfaces of the EngineConn Factory;
- Define the core abstraction and interfaces of Executor Manager.

### Checklist

- [x] I have read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests).
- [ ] I have explained the need for this PR and the problem it solves
- [ ] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [Linkis mailing list](https://linkis.apache.org/community/how-to-subscribe) first)
- [ ] **If this is a code change**: I have written unit tests to fully verify the new behavior.

### CODE REVIEW

File: linkis-computation-governance/linkis-engineconn-manager/linkis-engineconn-manager-server/src/main/java/org/apache/linkis/ecm/restful/ECMRestfulApi.java [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+@Api(tags = "ECM")
+@RequestMapping(path = "/engineconnManager")
+@RestController
+public class ECMRestfulApi {
+
+  private final Logger logger = LoggerFactory.getLogger(ECMRestfulApi.class);
+
+  @RequestMapping(path = "/downloadEngineLog", method = RequestMethod.POST)
+  public void downloadEngineLog(
```
Comment (casionone) [inline linkis-computation-governance/linkis-engineconn-manager/linkis-engineconn-manager-server/src/main/java/org/apache/linkis/ecm/restful/ECMRestfulApi.java:61]: Add swagger annotation

File: linkis-computation-governance/linkis-engineconn-manager/linkis-engineconn-manager-server/src/main/java/org/apache/linkis/ecm/restful/ECMRestfulApi.java [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+            "attachment;filename=" + instance.replace(":", "_") + "_" + logType + ".txt");
+        outputStream = response.getOutputStream();
+        while ((bytesRead = fis.read(buffer, 0, 1024)) != -1) {
+          outputStream.write(buffer, 0, bytesRead);
+        }
+      } catch (IOException e) {
+        logger.error("download failed", e);
+        response.reset();
+        response.setCharacterEncoding(Consts.UTF_8.toString());
+        response.setContentType("text/plain; charset=utf-8");
```
Comment (casionone) [inline linkis-computation-governance/linkis-engineconn-manager/linkis-engineconn-manager-server/src/main/java/org/apache/linkis/ecm/restful/ECMRestfulApi.java:118]: add header content-length

File: linkis-computation-governance/linkis-engineconn-manager/linkis-engineconn-manager-server/src/main/java/org/apache/linkis/ecm/errorcode/EngineconnServerErrorCodeSummary.java [hunk 3]
```diff
@@ -33,7 +33,8 @@ public enum EngineconnServerErrorCodeSummary implements LinkisErrorCode {
       11110,
       "the parameters of engineConnInstance and ticketId are both not exists.(engineConnInstance 和ticketId 的参数都不存在.)"),
   LOG_IS_NOT_EXISTS(11110, "Log directory {0} does not exists.(日志目录 {0} 不存在.)"),
-  FAILED_TO_DOWNLOAD(911115, "failed to downLoad(下载失败)");
+  FAILED_TO_DOWNLOAD(911115, "failed to downLoad(下载失败)"),
+  FILE_IS_OVERSIZE(911116, "Download file has exceeded 100MB(下载文件已经超过100M)");
```
Comment (casionone) [inline linkis-computation-governance/linkis-engineconn-manager/linkis-engineconn-manager-server/src/main/java/org/apache/linkis/ecm/errorcode/EngineconnServerErrorCodeSummary.java:37]:
```
 FILE_IS_OVERSIZE(911116, "Download file size has exceeded 100MB(下载文件已超过100M)");
```

File: linkis-computation-governance/linkis-engineconn-manager/linkis-engineconn-manager-server/src/main/java/org/apache/linkis/ecm/restful/ECMRestfulApi.java (hunk 2 above)
Comment (v-kkhuang) [reply linkis-computation-governance/linkis-engineconn-manager/linkis-engineconn-manager-server/src/main/java/org/apache/linkis/ecm/restful/ECMRestfulApi.java:118]: done

File: linkis-computation-governance/linkis-engineconn-manager/linkis-engineconn-manager-server/src/main/java/org/apache/linkis/ecm/restful/ECMRestfulApi.java (hunk 1 above)
Comment (v-kkhuang) [reply linkis-computation-governance/linkis-engineconn-manager/linkis-engineconn-manager-server/src/main/java/org/apache/linkis/ecm/restful/ECMRestfulApi.java:61]: done

File: linkis-computation-governance/linkis-engineconn-manager/linkis-engineconn-manager-server/src/main/java/org/apache/linkis/ecm/errorcode/EngineconnServerErrorCodeSummary.java (hunk 3 above)
Comment (v-kkhuang) [reply linkis-computation-governance/linkis-engineconn-manager/linkis-engineconn-manager-server/src/main/java/org/apache/linkis/ecm/errorcode/EngineconnServerErrorCodeSummary.java:37]: done

File: linkis-computation-governance/linkis-engineconn-manager/linkis-engineconn-manager-server/src/main/java/org/apache/linkis/ecm/restful/ECMRestfulApi.java [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+        fis = new BufferedInputStream(inputStream);
+        byte[] buffer = new byte[1024];
+        int bytesRead = 0;
+        response.setCharacterEncoding(Consts.UTF_8.toString());
+        java.nio.file.Path source = Paths.get(inputFile.getPath());
+        response.addHeader("Content-Type", Files.probeContentType(source));
+        response.addHeader(
+            "Content-Disposition",
+            "attachment;filename=" + instance.replace(":", "_") + "_" + logType + ".txt");
+        response.addHeader("Content-Length", fileSizeInBytes + "");
```
Comment (casionone) [inline linkis-computation-governance/linkis-engineconn-manager/linkis-engineconn-manager-server/src/main/java/org/apache/linkis/ecm/restful/ECMRestfulApi.java:120]: add annotation for filename example
// filename eg:xxxxx
val filename= instance.replace(":", "_") + "_" + logType + ".txt"
.....

Comment (v-kkhuang) [reply linkis-computation-governance/linkis-engineconn-manager/linkis-engineconn-manager-server/src/main/java/org/apache/linkis/ecm/restful/ECMRestfulApi.java:120]: done

**Pull request merged**